api/v1beta1/modulebuildsignconfig_types.go

```diff
@@ -17,9 +17,51 @@ limitations under the License.
 package v1beta1
 
 import (
+	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+type BuildOrSignAction string
+
+const (
+	// BuildImage means that image needs to be built
+	BuildImage BuildOrSignAction = "BuildImage"
+
+	// SignImage means that image needs to be built
+	SignImage BuildOrSignAction = "SignImage"
+
+	// ImageBuildFailed means that image does not exists and the Build failed
+	ImageBuildFailed ImageState = "BuildFailed"
+
+	// ImageBuildSucceeded means that image has been built and pushed succesfully
+	ImageBuildSucceeded ImageState = "BuildSucceeded"
+
+	// ImageSignFailed means that image does not exists and the Sign failed
+	ImageSignFailed ImageState = "SignFailed"
+
+	// ImageSignSucceeded means that image has been signed and pushed succesfully
+	ImageSignSucceeded ImageState = "SignSucceeded"
+)
+
+// ModuleImageSpec describes the image whose state needs to be queried
+type ModuleBuildSignSpec struct {
+	ModuleImageSpec `json:",inline"`
+
+	// Action to take on the image, either build or sign
```

Comment:
Let's have a kubebuilder validation to make sure it is only `Build` or `Sign` instead of a comment?